collections/item.go

```diff
@@ -2,19 +2,29 @@ package collections
 
 import (
 	"context"
-
-	storetypes "github.com/cosmos/cosmos-sdk/store/types"
 )
 
 // NewItem instantiates a new Item instance, given the value encoder of the item V.
-func NewItem[V any](sk storetypes.StoreKey, prefix Prefix, valueCodec ValueCodec[V]) Item[V] {
-	return (Item[V])(NewMap[noKey, V](sk, prefix, noKey{}, valueCodec))
+// Name and prefix must be unique within the schema and name must match the format specified by NameRegex, or
+// else this method will panic.
+func NewItem[V any](schema Schema, prefix Prefix, name string, valueCodec ValueCodec[V]) Item[V] {
```

Comment:
is `Prefix` here different than `schema.getPrefix()`? ditto on `name`.

Reply:
Ohhh, you're passing in a `Schema` object, which at the time of calling `NewItem`, doesn't know about it and thus adds it to the passed in `Schema`?

Reply:
Exactly